src/middleware/messageStore.js

```diff
@@ -253,7 +257,11 @@ export function setFinalStatus(ctx, callback) {
           ctx.response.status <= 299 &&
           routeSuccess
         ) {
-          tx.status = transactionStatus.SUCCESSFUL
+          if(ctx?.matchingChannel?.isAsynchronousProcess || !!ctx?.authorisedChannel?.isAsynchronousProcess){
+            tx.status = transactionStatus.PENDING_ASYNC
```

Comment:
```suggestion
          if(isAsynchronousProcess){
            tx.status = transactionStatus.PENDING_ASYNC
```

And please don't forget to run `npm run lint`  and `npm run lint:fix` to lint these changes